### DISCUSSION TITLE
Replicating emqx cluster and updatable bootstrapconfig

### DISCUSSION THREAD

**Discussion by alexa22013** [emqx/emqx-operator#710] [Q&A]
We have our emqx cluster deployed on gke cluster. All config required were add to bootstrap config on initial deployment, and whenever change in config is required its done via dashboard since bootstrap config cannot be updated

I understand bootstrap is not meant to be editable. But let's say I want to move my emqx cluster from one gke cluster to another or replicate an emqx cluster with the same config as existing cluster config, how do I do it?. Usually for any other apps, we simply copy the existing kubernetes yaml files since all config are maintained in the yaml itself and its the single source of config. But when it comes to emqx it becomes difficult to replicate the cluster since updates to config are done via dashboard and there becomes multiple sources of configs. 

I want to know if there is already a solution for this. If not, I believe it would really be helpful for the emqx cluster deployed on kubernetes to have an editable bootstrap config. And any config changes to go as rolling updates without bringing down the entire cluster and with no downtime which I believe the current emqx cluster have to do if config needs to be changed via bootstrap (delete and recreate the cluster)

**Comment by Rory-Z** [comment]
Hi @alexa22013 You can merge `data/cluster-overwrite.conf` from the EMQX container into `.spec.bootstrapConfig` and deploy a new cluster.
I'm sorry that we don't have a better way to deal with it now. Please look forward to the subsequent version of EMQX.
